File: base/grafite/.cache/lang/ts/lead.ts
class Lead {
    private thickness: number;
    private hardness: string;
    private size: number;
    
    
    public constructor(thickness: number, hardness: string, size: number) {
    }
    public usagePerSheet(): number {
        return 0;
    }
    public toString(): string {
        return `${this.thickness}:${this.hardness}:${this.size}`;
    }
    
    public getThickness(): number {
        return 0;
    }
    public getSize(): number {
        return 0;
    }
    public getHardness(): string {
        return "";
    }
    public setSize(value: number): void {
    }
    public setHardness(value: string): void {
    }
    public setThickness(value: number): void {
    }
}

export { Lead };

File: base/grafite/.cache/draft/java/Lead.java
import java.text.DecimalFormat;

public class Lead {
    private float thickness; //calibre
    private String hardness; //dureza
    private int size; //tamanho em mm

    public Lead(float thickness, String hardness, int size) {
    }

    public float getThickness() {
        return 0.0f;
    }

    public String getHardness() {
        return "";
    }

    public int getSize() {
        return 0;
    }

    public void setSize(int size) {
    }

    public int usagePerSheet() {
        if(hardness.equals("HB")) {
            return 1;
        } else if(hardness.equals("2B")) {
            return 2;
        } else if(hardness.equals("4B")) {
            return 4;
        } else {
            return 6;
        }
    }

    public String toString() {
        DecimalFormat form = new DecimalFormat("0.0");
        return form.format(thickness) + ":" + hardness + ":" + size;
    }
}
